libtenzir/src/execution_node.cpp

```diff
@@ -894,8 +894,12 @@ auto exec_node(
   self->state.metrics->values.operator_name = self->state.op->name();
   self->state.metrics->values.inbound_measurement.unit
     = operator_type_name<Input>();
+  self->state.metrics->values.inbound_measurement.internal
+    = std::is_void_v<Input> && self->state.op->internal();
   self->state.metrics->values.outbound_measurement.unit
     = operator_type_name<Output>();
+  self->state.metrics->values.outbound_measurement.internal
+    = std::is_void_v<Output> && self->state.op->internal();
```

Comment:
> [...] so that the flow between operators is classified as internal

That was my idea with this check. Your version makes a lot more sense then mine, so I'll change to this.

The alternative would've been adding two virtual functions to the operator API, which I thought was a bit unnecessary.